subsys/net/lib/openthread/platform/radio.c

```diff
@@ -1019,23 +1019,113 @@ uint8_t otPlatRadioGetCslAccuracy(otInstance *aInstance)
 }
 
 #if defined(CONFIG_OPENTHREAD_LINK_METRICS)
+/**
+ * +---------------------------------+----------------------+---------------------------------+-------------------+-------------------+                        
+ * | Length    | Element ID | Type=0 |      Vendor OUI      |                  Vendor Specific Information                            |
+ * +-----------+------------+--------+----------------------+---------------------------------+-------------------+-------------------+                        
+ * |           Bytes: 0-1            |          2-4         |                5                |         6         |   7 (optional)    |                       
+ * +-----------+---------------------+----------------------+---------------------------------+-------------------+-------------------+                        
+ * | Bits: 0-6 |    7-14    |   15   | IE_VENDOR_THREAD_OUI | IE_VENDOR_THREAD_ACK_PROBING_ID | LINK_METRIC_TOKEN | LINK_METRIC_TOKEN |
+ * +-----------+------------+--------+----------------------|---------------------------------|-------------------|-------------------|
+ */
+static uint8_t set_vendor_ie_header(bool lqi, bool link_margin, bool rssi, uint8_t *ie_header)
+{
+	const uint8_t IE_VENDOR_ID = 0x00; ///< Vendor-specific IE identifier
+	const uint8_t IE_VENDOR_THREAD_ACK_PROBING_ID =
+		0x00; ///< Thread Vendor-specific ACK Probing IE subtype ID
```

Comment:
I'd rather see comment on top, and code in a single line, if they don't fit together.